app/models/user_group.rb

```diff
@@ -14,4 +14,17 @@ def self.top_user_groups_by_date(start_date, end_date)
       .pluck("user_groups.group", "count(survey_user_groups.user_group_id)")
       .map { |user_group, total| { group_name: user_group, total: total } }
   end
+
+  def self.top_user_groups_for_page(page, start_date, end_date)
+    date_range = start_date..end_date
+
+    UserGroup
+      .joins(survey_user_groups: [{ survey: [{ survey_visit: [{ visit: :page_visits }] }] }])
+      .where("page_visits.page_id" => page.id)
+      .where("surveys.started_at" => date_range)
```

Comment:
Assume these are on separate lines for clarity.

Reply:
👍